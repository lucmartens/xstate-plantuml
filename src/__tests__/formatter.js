const format = require('../formatter').default;

const machine = [
  'machine',
  { key: 'users-crud' },
  ['action', ['machine', 'onEntry', 'a1']],
  ['state', ['on', 'users-crud_on']],
  ['state', ['off', 'users-crud_off']],
  ['event', ['[*]', 'users-crud_on']],
  ['event', ['users-crud_on', 'users-crud_off', 'TOGGLE']],
  ['event', ['users-crud_off', 'users-crud_on', 'TOGGLE']]
];

describe('formatter', () => {
  test('simple machine', () => {
    expect(format(['root', {}, machine])).toMatchSnapshot();
  });

  test('nested machine', () => {
    expect(format(['root', {}, [...machine, machine]])).toMatchSnapshot();
  });
});
