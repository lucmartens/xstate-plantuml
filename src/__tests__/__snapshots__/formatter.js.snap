// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatter nested machine 1`] = `
"@startuml
    state \\"name\\" as key {
        machine : onEntry/a1
        state \\"on\\" as users-crud_on
        state \\"off\\" as users-crud_off
        [*] --> users-crud_on 
        users-crud_on --> users-crud_off : TOGGLE
        users-crud_off --> users-crud_on : TOGGLE
        state \\"name\\" as key {
            machine : onEntry/a1
            state \\"on\\" as users-crud_on
            state \\"off\\" as users-crud_off
            [*] --> users-crud_on 
            users-crud_on --> users-crud_off : TOGGLE
            users-crud_off --> users-crud_on : TOGGLE
        }
    }
@enduml
"
`;

exports[`formatter simple machine 1`] = `
"@startuml
    state \\"name\\" as key {
        machine : onEntry/a1
        state \\"on\\" as users-crud_on
        state \\"off\\" as users-crud_off
        [*] --> users-crud_on 
        users-crud_on --> users-crud_off : TOGGLE
        users-crud_off --> users-crud_on : TOGGLE
    }
@enduml
"
`;
